daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -156,14 +163,20 @@ private[lf] object Speedy {
          Triggers. It is safe to use on ledger for off ledger code but
          not the other way around.
        */
-      val onLedger: Boolean,
+      val ledgerMode: LedgerMode,
   ) {
 
     /* kont manipulation... */
 
     @inline
     private[speedy] def kontDepth(): Int = kontStack.size()
 
+    private[lf] def withOnLedger[T](op: String)(f: OnLedger => T): T =
```

Comment:
Not quite sure if this callback-style approach makes that much sense here. The alternative would be a `getOnLedger` or something like that.